builder/azure/arm/template_factory.go

```diff
@@ -200,6 +201,29 @@ func GetVirtualMachineTemplateBuilder(config *Config) (*template.TemplateBuilder
 		}
 	}
 
+	if config.SecureBootEnabled || config.VTpmEnabled || config.EncryptionAtHost != nil {
+		switch config.SecurityType {
+		case constants.TrustedLaunch:
+			trustedlaunch := hashiVMSDK.SecurityTypesTrustedLaunch
+			chosenSecurityType = &trustedlaunch
+		case constants.ConfidentialVM:
+			confidentialVM := hashiVMSDK.SecurityTypesConfidentialVM
+			chosenSecurityType = &confidentialVM
+		case "":
+			if config.EncryptionAtHost != nil {
+				chosenSecurityType = nil
+				break
+			}
+			fallthrough
+		default:
+			return nil, fmt.Errorf("template: invalid security type %s. Valid options are 'TrustedLaunch' or 'ConfidentialVM'", config.SecurityType)
+		}
+		err = builder.SetSecurityProfile(config.SecureBootEnabled, config.VTpmEnabled, config.EncryptionAtHost, chosenSecurityType)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
The way we handle security type is a bit confusing here to me, as it's not clear to the user when a security type is required,  can we refactor this to not use break/fallthrough and make this error case a bit clearer

Reply:
Yes I agree with this. I will refactor this code. Also I discovered this am that this is not being evaluated when trying to create a VHD (to a storage account) from a CVM source. The refactor will include support for this.

Reply:
Refactored in [617b8d3](https://github.com/hashicorp/packer-plugin-azure/pull/373/commits/617b8d33cb53b8e75e3146043d4c3fa536cc60c4) ✅